app/src/main/java/ch/sdp/vibester/BuzzerScoreUpdater.kt

```diff
@@ -24,15 +27,31 @@ class BuzzerScoreUpdater(ids: ArrayList<Int>, scores: Array<Int>) {
         return buzzerToScoreMap
     }
 
+    fun getWinnerId(): Int {
+        return winnerId
+    }
+
     /**
      * updates the score corresponding to the pressed buzzer
      * checks first if the given id is a buzzer id (and not NO_BUZZER_PRESSED)
      */
-    fun updateScoresArray(id: Int) {
+    fun updateScoresArray(id: Int, point: Int) {
         if (!buzzerToScoreMap.keys.contains(id)) {
             return
         }
-        val updatedScore = buzzerToScoreMap.getOrDefault(id, 0) + 1 // should never get to default,
+        val updatedScore = max(buzzerToScoreMap.getOrDefault(id, 0) + point, 0) // should never get to default

```

Comment:
Did you put max considering that the score can be negative or what was the reason?

Reply:
Oh, I added a functionality that gives the player a -1 penalty if they buzz and guess wrong (instead of keeping their score unchanged). If they guess wrong when their score is 0, the score should not become negative.